test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java

```diff
@@ -77,29 +91,63 @@ public ClientYamlTestResponse callApi(String apiName, Map<String, String> params
         }
     }
 
-    private String actualBody(List<Map<String, Object>> bodies) throws IOException {
+    private HttpEntity createEntity(List<Map<String, Object>> bodies, Map<String, String> headers) throws IOException {
         if (bodies.isEmpty()) {
-            return "";
+            return null;
         }
-
-        if (bodies.size() == 1) {
-            return bodyAsString(stash.replaceStashedValues(bodies.get(0)));
+        XContentType xContentType = null;
+        String contentType = headers.get("Content-Type");
```

Comment:
I think we may want to document that the headers used in the yaml tests are case sensitive or fix the map to be case insensitive?